### PR TITLE
Default value support in protocol

### PR DESCRIPTION
**Pull request by anandsubbu007**
made both old and this new format valid

Yaml file in protocol
```
fields:
  name: 
    type: String
    defaultvalue: 'A'
  listData: 
    type: List<String>
    defaultvalue: '[]'
  intType:
    type: int
    defaultvalue: 0
  intTypewithApi:
    type: int
    scope: api
    defaultvalue: 0
  emptyint: int
  emptyintwithApi: int,  api
  emptyStrdata: String
  emptyintwithDatabase: int,  database
```
SQL Syntax
```
CREATE TABLE example (
  "id" serial,
  "name" text NOT NULL DEFAULT A,
  "listData" json NOT NULL DEFAULT '[]',
  "intType" integer NOT NULL DEFAULT '0',
  "emptyint" integer NOT NULL,
  "emptyStrdata" text NOT NULL,
  "emptyintwithDatabase" integer NOT NULL
);

ALTER TABLE ONLY example
  ADD CONSTRAINT example_pkey PRIMARY KEY (id);
```
In Class Model constructor
```
  Example({
    this.id,
    this.name = 'A',
    this.listData = const [],
    this.intType = 0,
    this.intTypewithApi = 0,
    required this.emptyint,
    required this.emptyintwithApi,
    required this.emptyStrdata,
    required this.emptyintwithDatabase,
  });
```

### CODE REVIEW

**Comment by vlidholt** [comment]
Does this work with all primitives, e.g. DateTime, ByteData, String, bool, int, double? Does it work with SerializableObjects (this may not be a requirement)? Your examples only have empty arrays, does it also work with arrays with values? Any updates to the serialization also needs tests.

File: tools/serverpod_cli/bin/generator/pgsql_generator.dart [hunk 1]
```diff
@@ -34,6 +34,10 @@ class PgsqlGenerator {
     out += '--\n';
     out += '\n';
 
+    // Todo: Drop Table If Already Exist. But all data will be erased
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/generator/pgsql_generator.dart:37]: I cannot approve PRs with TODOs. Everything in the PR needs to be functional.

**Comment by anandsubbu007** [comment]
i think that in future we can also define link between tables

**Comment by vlidholt** [comment]
It looks like this pull request is pulled from dev branch but sent to the main branch, this makes it very hard to see the relevant changes as it also includes all changes on the dev branch.

**Pull request closed without merging**